scientific_library/tvb/simulator/monitors.py

```diff
@@ -174,19 +179,43 @@ class Raw(Monitor):
         required=False)
     # order = -1
 
-    def config_for_sim(self, simulator):
+    def _config_vois(self, simulator):
+        self.voi = numpy.arange(len(simulator.model.variables_of_interest))
+
+    def _config_time(self, simulator):
+        self.dt = simulator.integrator.dt
         if self.period != simulator.integrator.dt:
             self.log.debug('Raw period not equal to integration time step, overriding')
         self.period = simulator.integrator.dt
-        super(Raw, self).config_for_sim(simulator)
         self.istep = 1
-        self.voi = numpy.arange(len(simulator.model.variables_of_interest))
 
     def sample(self, step, state):
         time = step * self.dt
         return [time, state]
 
 
+class RawVoi(Raw):
+    """
+        A monitor that records the output raw data from a tvb simulation:
+        It collects:
+
+            - selected state variables of all modes from class :Model:
+            - all nodes of a region or surface based
+            - all the integration time steps
+
+        """
+
+    _ui_name = "Selected State Variables' Raw recording"
+
+    def _config_vois(self, simulator):
+        self.voi = self.variables_of_interest
+        if self.voi is None or self.voi.size == 0:
+            self.voi = numpy.r_[:len(simulator.model.variables_of_interest)]
+
+    def sample(self, step, state):
+        return super(RawVoi, self).sample(step, state[self.voi])
```

Comment:
Python3 lets you write just `super().sample()` now, which helps avoid errors with copy paste.

Reply:
There are more super() calls there, so, I would suggest that we do a dedicated correction to a larger part of the code if we wish to change this.

Reply:
I agree